packages/base/src/3dview/mainview.tsx

```diff
@@ -1110,6 +1114,24 @@ export class MainView extends React.Component<IProps, IStates> {
         this._updateClipping();
       }
     }
+
+    if (change.key === 'wireframe') {
+      const wireframeEnabled = change.newValue as boolean | undefined;
+
+      if (wireframeEnabled !== undefined) {
+        this.setState({ wireframe: wireframeEnabled }, () => {
```

Comment:
Setting the state like that will erase all other states. Using a function with an old state would be better.

Reply:
On it, Thanks for suggesting.